server/src/main/java/org/opensearch/gateway/remote/routingtable/IndexRoutingTableInputStream.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.routingtable;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.IndexShardRoutingTable;
+import org.opensearch.common.io.stream.BufferedChecksumStreamOutput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.bytes.BytesReference;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+
+public class IndexRoutingTableInputStream extends InputStream {
```

Comment:
This is a useful extension. Just thinking if this can be made generic where, in this case, T would be `IndexRoutingTable`?